File: Programmers/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;
bool comp(string a, string b)
{
    if(a + b > b + a)
        return true;
    else return false;
}

string solution(vector<int> numbers) {
    string answer = "";
    vector<string> s;
    for(int i : numbers)
        s.push_back(to_string(i));
    sort(s.begin(), s.end(), comp);
    
    for(string i : s)
        answer += i;
    if(answer[0]=='0')
        answer = '0';
    return answer;
}

File: Programmers/가장 큰 수.java
import java.util.*;
class Solution {
    public String solution(int[] numbers) {
        String[] nums = new String[numbers.length];
        
        for(int i = 0; i < numbers.length; i++)
            nums[i] = numbers[i] + "";
        
        Arrays.sort(nums, new Comparator<String>() {
            public int compare(String o1, String o2){
                return (o2 + o1).compareTo(o1 + o2);
            }
        });
        String answer = "";
        for(int i = 0; i < nums.length; i++)
            answer += nums[i];
        
        return answer.charAt(0) == '0' ? "0" : answer;
    }
}